unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/CharacterPreview/PreviewCameraRotation.cs

```diff
@@ -19,8 +21,13 @@ public class PreviewCameraRotation : MonoBehaviour, IDragHandler, IBeginDragHand
         private Coroutine slowDownCoroutine;
 
         private float timer;
+        private DateTime startDragDateTime;
 
-        public void OnBeginDrag(PointerEventData eventData) { AudioScriptableObjects.buttonClick.Play(true); }
+        public void OnBeginDrag(PointerEventData eventData)
+        {
+            startDragDateTime = DateTime.Now;
```

Comment:
I don't really like how analytics pierces random places in the code making it holey.
In my opinion it should be designed in a completely detachable and separate way.
Maybe we can think of a general approach to it?

Reply:
Yeah I understand, we should standardise this as well, it is quite common that analytics "dirty" a bit the code with their logic, should we create even for this a refactor task to isolate all the analytics logic?

Reply:
Can you create a task for the specification of analytics decoupling? If will be super helpful if you add more as many thoughts and ideas as possible